### PR TITLE
PICARD-1383: Select release with matching barcode on disc ID lookup

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem
When a disc ID lookup contains multiple results picking the correct one can be tricky. We can assist the user by making use of the barcode stored on the disc itself.

* JIRA ticket (_optional_): [PICARD-1383](https://tickets.metabrainz.org/browse/PICARD-1383)

# Solution
Read the MCN field (containing the EAN as per spec) from the disc when generating the disc ID. Use this barcode to pre-select the release that matches the barcode.

# Action

### CODE REVIEW

File: picard/ui/cdlookup.py [hunk 1]
```diff
@@ -61,11 +64,15 @@ def __init__(self, releases, disc, parent=None):
             def myjoin(l):
                 return "\n".join(l)
 
+            selected = None
             for release in self.releases:
                 labels, catalog_numbers = label_info_from_node(release['label-info'])
                 dates, countries = release_dates_and_countries_from_node(release)
                 barcode = release['barcode'] if "barcode" in release else ""
                 item = QtWidgets.QTreeWidgetItem(self.ui.release_list)
+                print(disc.mcn, barcode, barcode == disc.mcn)
```
Comment (zas) [inline picard/ui/cdlookup.py]: Debug ?

File: picard/util/__init__.py [hunk 2]
```diff
@@ -566,3 +566,21 @@ def reconnect(signal, newhandler=None, oldhandler=None):
...
+    """
+    if barcode1 == barcode2:
+        return True
+    if not barcode1 or not barcode2:
+        return False
+    if len(barcode1) == 12:
+        barcode1 = '0' + barcode1
+    if len(barcode2) == 12:
+        barcode2 = '0' + barcode2
+    return barcode1 == barcode2
```
Comment (zas) [inline picard/util/__init__.py]: You may simplify by 0 padding both strings (to 13, for EAN13) using:

```python
return barcode1.zfill(13) == barcode2.zfill(13)
```

Then you can drop `if len...` tests.

Comment (phw) [reply picard/util/__init__.py]: Much better. Thanks, fixed

File: test/test_utils.py [hunk 3]
```diff
@@ -224,3 +224,18 @@ def test_invalid_png_data(self):
...
                           imageinfo.identify, data)
+
+class CompareBarcodesTest(unittest.TestCase):
+
+    def test_same(self):
+        self.assertTrue(util.compare_barcodes('0727361379704', '0727361379704'))
+        self.assertTrue(util.compare_barcodes('727361379704', '727361379704'))
+        self.assertTrue(util.compare_barcodes('727361379704', '0727361379704'))
+        self.assertTrue(util.compare_barcodes('0727361379704', '727361379704'))
+        self.assertTrue(util.compare_barcodes(None, None))
```
Comment (zas) [inline test/test_utils.py]: I'd add tests for empty strings too:

```python
self.assertTrue(util.compare_barcodes('', ''))
```

File: test/test_utils.py [hunk 4]
```diff
@@ -224,3 +224,18 @@ def test_invalid_png_data(self):
...
+
+    def test_not_same(self):
+        self.assertFalse(util.compare_barcodes('0727361379704', '0727361379705'))
+        self.assertFalse(util.compare_barcodes('727361379704', '1727361379704'))
+        self.assertFalse(util.compare_barcodes('0727361379704', None))
+        self.assertFalse(util.compare_barcodes(None, '0727361379704'))
```
Comment (zas) [inline test/test_utils.py]: I'd add tests for empty strings too:

```python
self.assertFalse(util.compare_barcodes(None, ''))
```

Comment (phw) [reply test/test_utils.py]: Is this actually what we would expect, or shouldn't `None` and `''` be considered equal?

Comment (zas) [reply test/test_utils.py]: I guess it'd make sense.

Comment (phw) [reply test/test_utils.py]: Yes, I think it makes most sense. I implemented it this way, which needed some update to the compare_barcodes function itself.
